### DISCUSSION TITLE
Fallback/fail-safe storage arrangement? IE if encrypted or network storage location is  unavailable.

### DISCUSSION THREAD

**Discussion by Mysterylectricity** [blakeblackshear/frigate#9385] [Ask A Question]
Before I suggest this a a feature, I thought I'd ask if this functionality is already built in. 

When I set about to use zoneminder with a manually-mounted encrypted drive I thought I'd have no end of trouble. But I was delighted to find it handled  the situation elegantly by default. 

The default configuration stores in /var/cache/zoneminder/events. I dedicated about 600G of my 1T NVME boot drive to /var in anticipation. 

I then dedicated 4T of my 12T HDD as a LUKS encrypted drive for ZM, added it to the list of storage locations, and specified this volume as the storage location for each of my cameras.

As it turns out, If there's an unexpected reboot or I forget to unlock the LUKS drive, ZM will happily write to the SSD until the LUKS volume is unlocked. It's almost seamless, though it does log some complaints in the  uncommon case where an unlocked drive gets subsequently unmounted. But only initially. 

Once the LUKS volume is unlocked, it starts writing to that volume again. 

Obviously the database is on the SSD.

Purists might argue ZM shouldn't do this unless it's explicitly told, ie: via a list associated with each camera specifying the preferred order of locations. But it works for me.

Does frigate have a similar ability?

If not, doesn't it sound handy?

**Comment by NickM-27** [comment]
You may be able to set something up like that in docker

**Reply by Mysterylectricity** [reply]
Good point given that the storage area is defined in the docker .yaml. I'll look into that.
